static/js/components/forms/validation.ts

```diff
@@ -0,0 +1,32 @@
+import * as yup from "yup"
+import { SchemaOf, setLocale } from "yup"
+
+import { ConfigItem } from "../../types/websites"
+
+// This is added to properly handle file fields, which can have a "null" value
+setLocale({
+  mixed: {
+    notType: "${path} is required"
+  }
+})
+
+const defaultYupField = yup.string()
+const widgetToYupMap = {
+  string:   yup.string(),
+  text:     yup.string(),
+  markdown: yup.string(),
+  file:     yup.object().shape({
+    name: yup.string().required()
+  })
+}
+
+export const getContentSchema = (configItem: ConfigItem): SchemaOf<any> => {
+  const yupObjectShape = {}
+  configItem.fields.forEach(field => {
+    if (field.required) {
```

Comment:
I think we should set the schema for each field even if it's not required. I think the only difference is the `.required()` function call